src/components/general/Menu/index.tsx

```diff
@@ -0,0 +1,117 @@
+import * as React from "react";
+import classNames from "classnames";
+import useElevationClassName, { Elevation } from "../../../hooks/useElevationClassName";
+import useKeyboardNavigation from "../../../hooks/useKeyboardNavigation";
+import MenuItem, { MenuItemType } from "./MenuItem";
+import styles from "./styles.less";
+
+export type MenuSection = {
+    key: string,
+    title?: string,
+    items: MenuItemType[],
+};
+
+export type MenuProps = {
+    sections: MenuSection[],
+    elevation?: Elevation,
+    onClick?: (item: MenuItemType) => void | Promise<void>,
+    keyboardNavigationRef?: HTMLElement | null,
+};
+
+const Menu: React.FC<MenuProps> = props => {
+    const [focusedSectionKey, setFocusedSectionKey] = React.useState<string | null>(props.sections.length ? props.sections[0].key : null);
+    const [focusedItemKey, setFocusedItemKey] = React.useState<string | null>(props.sections.length && props.sections[0].items.length ? props.sections[0].items[0].key : null);
+
+    const nextOrPrev = (direction: number) => {
+        const sectionIndex = props.sections.findIndex(section => section.key === focusedSectionKey);
+        let nextSectionIndex = sectionIndex;
+
+        if (sectionIndex === -1) {
+            nextSectionIndex = direction === -1 ? props.sections.length - 1 : 0;
+        }
+
+        const items = props.sections[nextSectionIndex].items;
+
+        const itemIndex = items.findIndex(item => item.key === focusedItemKey);
+        let nextItemIndex = itemIndex;
+
+        if (nextItemIndex === -1) {
+            nextSectionIndex = direction === -1 ? props.sections.length - 1 : 0;
+            nextItemIndex = direction === -1 ? props.sections[nextSectionIndex].items.length - 1 : 0;

```

Comment:
Could we use the items definition on line 33 here? 
`nextItemIndex = direction === -1 ? items.length - 1 : 0;`

Reply:
Here the next section index has changed and items defined on line 33 no longer refers to the items in the "next section"